### DISCUSSION TITLE
awsscvcommonsecrets | CREATE_FAILED | Invalid name. Must be a valid name containing alphanumeric characters, or any of the following: -/_+=.@!

### DISCUSSION THREAD

**Discussion by ShinobiSec** [amazon-connect/amazon-connect-salesforce-scv#78] [Q&A] [answered]
Carrying this into discussion since I am seeking help, I do not believe I have an error resulting from any code/resources

https://github.com/amazon-connect/amazon-connect-salesforce-scv/issues/76

I am trying to recreate the AWSSCV-SalesforceConfig stack in the region our instance is in. us-west-2. Any help getting past this issue would be greatly appreciated. 

I have had an unsuccessful AWSSCV-SalesforceConfig stack deployment. With error:
2021-12-25 00:06:42 UTC-0500 | AWSSCV-SalesforceConfig | ROLLBACK_COMPLETE | -
2021-12-25 00:06:42 UTC-0500 | awsscvcommonsecrets | DELETE_COMPLETE | -
2021-12-25 00:06:34 UTC-0500 | AWSSCV-SalesforceConfig | ROLLBACK_IN_PROGRESS | The following resource(s) failed to create: [awsscvcommonsecrets]. Rollback requested by user.
2021-12-25 00:06:33 UTC-0500 | awsscvcommonsecrets | CREATE_FAILED | Invalid name. Must be a valid name containing alphanumeric characters, or any of the following: -/_+=.@!
2021-12-25 00:06:31 UTC-0500 | awsscvcommonsecrets | CREATE_IN_PROGRESS | -
2021-12-25 00:06:27 UTC-0500 | AWSSCV-SalesforceConfig | CREATE_IN_PROGRESS | User Initiated

**Comment by ShinobiSec** [comment]
I have successfully deployed all stacks and resources. The validator test came back successful. Successfully leave a voicemail with published contact flows provided with updates required completed. 

The only issue is no new cases are being created in salesforce. What would be the best way to debug this issue?

**Reply by dcarbonetti** [reply]
If the validator was successful (and you checked inside the success message to see what it actually said as sometimes it looks good but isn't) the next step would be to debug is by looking at logs from the packager lambda or by looking at debug logs in salesforce.

A couple of things off the top of my head that can cause it are:
Record types on cases 
Validation rules on cases

**Reply by ShinobiSec** [reply]
Thank you for your guidance. I will be working on this again on Monday. Ill report back any findings if I hit a roadblock or see something that doesn't make sense. 

When I started working on this, I was concerned about this process using cases. We do have validation rules we are using with cases. I hope this is able to be remedied.

**Reply by ShinobiSec** [reply]
Packager error found in logs 
[ERROR]	2021-12-25T21:38:41.363Z	---    list indices must be integers or slices, not str

**Reply by ShinobiSec** [reply]
When running a test with default test values I receive this response. 
Response
{
  "errorMessage": "'Records'",
  "errorType": "KeyError",
  "stackTrace": [
    "  File \"/var/task/awsscv_vmx_packager.py\", line 35, in lambda_handler\n    for record in event['Records']:\n"
  ]
}

Function Logs
START RequestId: 43bf8c2b-daa1-4b48-a213-ada9e8548833 Version: $LATEST
[INFO]	2021-12-25T23:59:52.251Z	(ID-erased incase sensitive info) 	Found credentials in environment variables.
[ERROR] KeyError: 'Records'
Traceback (most recent call last):
  File "/var/task/awsscv_vmx_packager.py", line 35, in lambda_handler
    for record in event['Records']:END RequestId: (ID-erased incase sensitive info) 
REPORT RequestId: (ID-erased incase sensitive info) 	Duration: 819.86 ms	Billed Duration: 820 ms	Memory Size: 128 MB	Max Memory Used: 71 MB	Init Duration: 350.42 ms

Request ID
(ID-erased incase sensitive info)

**Reply by dougjaso** [reply]
Waht do you mean by "Default test values" can you include your test values here?

**Comment by ShinobiSec** [comment]
We use cases in our instance for unique purpose. Is there any documentation on perhaps creating a custom object where Voicemails can be recorded in our org and saved within?

**Reply by dougjaso** [reply]
The next release of Voicemail Express includes the option to write to Tasks instead of Cases. That version of the packager will be more modularized to make it easier to adapt to other use cases as well. Expected release is end of Jan.

**Comment by dougjaso** [comment, marked as answer]
Wrapped into [this case](https://github.com/amazon-connect/amazon-connect-salesforce-scv/discussions/81). Answer is there. Marking this as answered.

**Comment by naddyson0** [comment]
My problem was that I didn't build before deploying
